Sources/Workspace/DefaultPluginScriptRunner.swift

```diff
@@ -157,6 +157,16 @@ public struct DefaultPluginScriptRunner: PluginScriptRunner {
                 macOSPackageDescriptionPath = runtimePath.appending(component: "libPackagePlugin.dylib")
             }
 
+            // On macOS earlier than 12, add an rpath to the directory that contains the concurrency fallback library.
+            if #available(macOS 12.0, *) {
+                // Nothing is needed; the system has everything we need.
+            }
+            else {
+                // Add an `-rpath` so the Swift 5.5 fallback libraries can be found.
+                let swiftSupportLibPath = self.toolchain.swiftCompilerPath.parentDirectory.parentDirectory.appending(components: "lib", "swift-5.5", "macosx")
+                command += ["-Xlinker", "-rpath", "-Xlinker", swiftSupportLibPath.pathString]
+            }
```

Comment:
1. do we need to care about other darwin based system beyond macOS?
2. should we test for compiler version and not assume 5.5?

Reply:
That's a great point — I'll update the condition to include iOS 15 at least (I don't know if tvOS or watchOS are interesting).  Also, that seems like perhaps a problem with this PR in general, because we only want to add `-rpath` on Darwin on macOS, not on non-Darwin.  I'll revise.

Reply:
I guess the original reason I did just macOS here is that the entire function is full of `#if macOS()` so will in practice not work on other Darwin platforms.  Note that macOS is the only one that allows spawning subprocesses — this code won't work on iOS, tvOS, and watchOS.

Reply:
That's a good question.  @DougGregor would the compiler version for the async compatibility libraries be updated for the currently used compiler, i.e. should we base this on what version of the compiler happens to be installed?  Or would newer ones still use the `swift-5.5` subdirectory because the async feature was introduced in Swift 5.5?